### PR TITLE
pythonPackages.pyinstaller: fix darwin build

### LINKED ISSUE #342931: Build failure: Pyinstaller
### Steps To Reproduce

Steps to reproduce the behaviour:
1. build PyInstaller on Darwin

### Build log

```
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:9:12:
            8|
            9|   strict = derivationStrict drvAttrs;
             |            ^
           10|

       … while evaluating derivation 'nix-shell'
         whose name attribute is located at /nix/store/d9gbq853jvbccrz5g3y0irbwgc57w137-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'nix-shell'
         at /nix/store/d9gbq853jvbccrz5g3y0irbwgc57w137-source/pkgs/stdenv/generic/make-derivation.nix:380:7:
          379|       depsBuildBuild              = elemAt (elemAt dependencies 0) 0;
          380|       nativeBuildInputs           = elemAt (elemAt dependencies 0) 1;
             |       ^
          381|       depsBuildTarget             = elemAt (elemAt dependencies 0) 2;

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: Package ‘glibc-nolibgcc-2.39-52’ in /nix/store/d9gbq853jvbccrz5g3y0irbwgc57w137-source/pkgs/development/libraries/glibc/default.nix:183 is not available on the requested hostPlatform:
         hostPlatform.config = "aarch64-apple-darwin"
         package.meta.platforms = [
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "loongarch64-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mips-linux"
           "mips64-linux"
           "mips64el-linux"
           "mipsel-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.
```

### Additional context

It fails to build at gcc as gcc is not avlible on macOS 

### Notify maintainers

CC @h7x4 

### Metadata

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.23.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Eveeifyeve**
## Description of changes

Fixes: #342931 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
  - [x] `sandbox = false`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Eveeifyeve** [comment]
It's failing at `passthru.tests` as python is not configured for tkinter.

File: pkgs/development/python-modules/pyinstaller/default.nix [hunk 1]
```diff
@@ -34,10 +36,11 @@ buildPythonPackage rec {
   dependencies = [
     altgraph
     packaging
+    python312Packages.macholib
     pyinstaller-hooks-contrib
   ];
 
-  makeWrapperArgs = [
+  makeWrapperArgs = lib.optionals stdenv.isLinux [
```
Comment (h7x4) [inline pkgs/development/python-modules/pyinstaller/default.nix:43]:
```suggestion
  makeWrapperArgs = lib.optionals stdenv.hostPlatform.isLinux [
```

File: pkgs/development/python-modules/pyinstaller/default.nix [hunk 2]
```diff
@@ -34,10 +36,11 @@ buildPythonPackage rec {
   dependencies = [
     altgraph
     packaging
+    python312Packages.macholib
```
Comment (h7x4) [inline pkgs/development/python-modules/pyinstaller/default.nix:39]: Is there a reason this has to be python312Packages specifically? If not, I think we could just import it as normal

```suggestion
    macholib
```

**Comment by felixalbrigtsen** [comment]
Result of `nixpkgs-review pr 345199` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pyinstaller</li>
    <li>python311Packages.pyinstaller.dist</li>
    <li>python312Packages.pyinstaller</li>
    <li>python312Packages.pyinstaller.dist</li>
  </ul>
</details>

🚀

**Comment by Eveeifyeve** [comment]
Result of `nixpkgs-review pr 345199` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pyinstaller</li>
    <li>python311Packages.pyinstaller.dist</li>
    <li>python312Packages.pyinstaller</li>
    <li>python312Packages.pyinstaller.dist</li>
  </ul>
</details>

**Comment by Eveeifyeve** [comment]
> Result of `nixpkgs-review pr 345199` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 
> <details>
> 
>   <summary>4 packages built:</summary>
> 
>   <ul>
> 
>     <li>python311Packages.pyinstaller</li>
> 
>     <li>python311Packages.pyinstaller.dist</li>
> 
>     <li>python312Packages.pyinstaller</li>
> 
>     <li>python312Packages.pyinstaller.dist</li>
> 
>   </ul>
> 
> </details>
> 
> 
> 
> 🚀 

It works.

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 345199` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pyinstaller</li>
    <li>python311Packages.pyinstaller.dist</li>
    <li>python312Packages.pyinstaller</li>
    <li>python312Packages.pyinstaller.dist</li>
  </ul>
</details>

**Comment by wildwestrom** [comment]
Result of `nixpkgs-review pr 345199` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.pyinstaller</li>
    <li>python311Packages.pyinstaller.dist</li>
    <li>python312Packages.pyinstaller</li>
    <li>python312Packages.pyinstaller.dist</li>
  </ul>
</details>

**Comment by wildwestrom** [comment]
Result of `nixpkgs-review pr 345199` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.pyinstaller</li>
    <li>python311Packages.pyinstaller.dist</li>
    <li>python312Packages.pyinstaller</li>
    <li>python312Packages.pyinstaller.dist</li>
  </ul>
</details>

**Review by h7x4 — APPROVED** [review]

**Comment by wildwestrom** [comment]
Result of `nixpkgs-review pr 345199` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.pyinstaller</li>
    <li>python311Packages.pyinstaller.dist</li>
    <li>python312Packages.pyinstaller</li>
    <li>python312Packages.pyinstaller.dist</li>
  </ul>
</details>

**Comment by h7x4** [comment]
Was indeed a bit quick here, sorry about that @wildwestrom. I'll be opening a PR in a moment.

**Pull request merged**